libraries/AP_Mission/AP_Mission.h

```diff
@@ -532,13 +532,21 @@ class AP_Mission
     // find the nearest landing sequence starting point (DO_LAND_START) and
     // return its index.  Returns 0 if no appropriate DO_LAND_START point can
     // be found.
-    uint16_t get_landing_sequence_start() const;
+    uint16_t get_landing_sequence_start();
+
+    // find the shortest distance to landing via DO_LAND_START and
+    // return its index.  Returns 0 if no appropriate DO_LAND_START point can
+    // be found.
+    uint16_t get_shortest_landing_sequence_start();
```

Comment:
Do you have an example use case for this ? I see it as a weak duplicate from the rally point, but I can be wrong

Reply:
You can only go directly to a rally point, this is much much more powerful, allowing take a pre-defined route to get there. The current code just jumps to the closest do land start, even if that might be preceding massive mission leg, this new code accounts for the mission after the do land start, not the just distance to get to point its-self.

Reply:
That is a much better introduction to what this do IMHO ;-P